### PR TITLE
fix spatial indexing for vbar/hbar

### LINKED ISSUE #4924: Tooltips unavailable for `vbar` and `hbar` glyphs
## Expected behavior and the observed behavior
Using the hover tools on `vbar` and (I assume) `hbar` glyphs currently does not work.

## Example
via http://stackoverflow.com/questions/38750508/using-the-hover-tool-with-vbar-glyphs
```python 
import numpy
import pandas
from bokeh import charts, plotting, models
plotting.output_notebook()

blue = 'STEELBLUE'
green = 'FORESTGREEN'

datalist = [
    {'month': 'Oct', 'rain': 131., 'snow':  0.0, 'wy_month': 1},
    {'month': 'Nov', 'rain': 12.4, 'snow':  0.0, 'wy_month': 2},
    {'month': 'Dec', 'rain': 43.0, 'snow': 13.5, 'wy_month': 3},
    {'month': 'Jan', 'rain': 63.0, 'snow':  9.2, 'wy_month': 4},
    {'month': 'Feb', 'rain': 72.6, 'snow': 35.3, 'wy_month': 5},
    {'month': 'Mar', 'rain': 13.5, 'snow':  4.2, 'wy_month': 6},
    {'month': 'Apr', 'rain': 107., 'snow':  1.5, 'wy_month': 7},
    {'month': 'May', 'rain': 77.0, 'snow':  0.0, 'wy_month': 8},
    {'month': 'Jun', 'rain': 107., 'snow':  0.0, 'wy_month': 9},
    {'month': 'Jul', 'rain': 216., 'snow':  0.0, 'wy_month': 10},
    {'month': 'Aug', 'rain': 76.8, 'snow':  0.0, 'wy_month': 11},
    {'month': 'Sep', 'rain': 76.4, 'snow':  0.0, 'wy_month': 12}
]

data = pandas.DataFrame(datalist).assign(total=lambda df: df['rain'] + df['snow'])
source = plotting.ColumnDataSource(data)

tooltips = [
    ("month", "@month"),
    ("rain", "@rain"),
    ("snow", "@snow"),
]

hover_bar = models.HoverTool(tooltips=tooltips)
TOOLS_bar = [hover_bar, models.ResizeTool(), models.ResetTool()]

fig = plotting.figure(width=600, height=300, y_range=(0, 250),
                      x_range=data['month'].tolist(),
                      tools=TOOLS_bar)
fig.vbar(x='wy_month', bottom=0, top='rain', width=0.5, color=blue, source=source)
fig.vbar(x='wy_month', bottom='rain', top='total', width=0.5, color=green, source=source)
plotting.show(fig)
```
## Console output
N/A -- this is known to not be implemented

## System info
```
λ bokeh info
Python version      :  3.5.2 |Continuum Analytics, Inc.| (default, Jul  5 2016, 11:41:13) [MSC v.1900 64 bit (AMD64)]
IPython version     :  5.0.0
Bokeh version       :  0.12.1
BokehJS static path :  C:\Users\phobson\Miniconda3\envs\tahoe\lib\site-packages\bokeh\server\static
```

## Desired behavior
I would like to implement the necessary `_hit_test` and `_index_data` methods to allow tooltips via a `HooverTool` on `vbar` and `hbar` glyphs.

### PR DESCRIPTION
**Pull request by bryevdv**
issues: fixes #4924 #5035

### CODE REVIEW

**Comment by bryevdv** [comment]
screen shot 2016-08-24 at 4 16 58 pm

**Comment by bryevdv** [comment]
ping @phobson thanks for getting this started, only some minor changes were needed to finish it.

**Comment by phobson** [comment]
Woooo hoooo!

File: bokehjs/src/coffee/models/glyphs/quad.coffee [hunk 1]
```diff
@@ -25,16 +25,10 @@ class QuadView extends Glyph.View
 
     for i in [0...left.length]
       l = left[i]
-      if isNaN(l) or not isFinite(l)
```
Comment (birdsarah) [inline bokehjs/src/coffee/models/glyphs/quad.coffee]: @bryevdv can you comment on why this is nolonger / was never necessary.

**Comment by birdsarah** [comment]
Happy to merge, but open an issue to add tests?

File: bokehjs/src/coffee/models/glyphs/quad.coffee (hunk 1 above)
Comment (bryevdv) [inline bokehjs/src/coffee/models/glyphs/quad.coffee]: the original code checked for `NaN`/`isFinite` on four variables independently. But those properties are viral: if any of a, b, c, d satisfy them, then their sum also satisfies them.

File: bokehjs/src/coffee/models/glyphs/quad.coffee [hunk 2]
```diff
@@ -25,16 +25,10 @@ class QuadView extends Glyph.View
...
-        continue
       r = right[i]
-      if isNaN(r) or not isFinite(r)
-        continue
       t = top[i]
-      if isNaN(t) or not isFinite(t)
-        continue
       b = bottom[i]
-      if isNaN(b) or not isFinite(b)
+      if isNaN(l+r+t+b) or not isFinite(l+r+t+b)
```
Comment (bryevdv) [inline bokehjs/src/coffee/models/glyphs/quad.coffee]: here is the line that checks the sum of the values, instead of checking each value individually

Comment (birdsarah) [inline bokehjs/src/coffee/models/glyphs/quad.coffee]: Sorry I missed this - thanks!

**Pull request merged**